StageIsChanged.cls

```diff
@@ -0,0 +1,8 @@
+trigger StageIsChanged on Opportunity (before update) {
+  
+    for(Opportunity opt : trigger.new){
+        if(opt.StageName == 'Closed Won' || opt.StageName =='Closed Lost'){

```

Comment:
who we know that state is changed we are not comparing this with previous condition.